models/baseModels/tests/testLoyaltyProgram.spec.ts

```diff
@@ -0,0 +1,275 @@
+    collectionFactor: 0.5,
+    minimumTotalSpent: 3000 },
+];
+
+test('create test docs', async (t) => {
+  await fyo.doc.getNewDoc(ModelNameEnum.Item, itemData).sync();
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.Item, itemData.name),
+    `dummy item ${itemData.name}  exists`
+  );
+
+  await fyo.doc.getNewDoc(ModelNameEnum.Party, partyData).sync();
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.Party, partyData.name),
+    `dummy party ${partyData.name} exists`
+  );
+
+  await fyo.doc.getNewDoc(ModelNameEnum.Account, accountData).sync();
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.Account, accountData.name),
+    `dummy account ${accountData.name}  exists`
+  );
+});
+
+test('create a Loyalty Program document', async (t) => {
+  const loyaltyProgramDoc = fyo.doc.getNewDoc(
+    ModelNameEnum.LoyaltyProgram,
+    loyaltyProgramData
+  );
+
+  loyaltyProgramDoc.collectionRules = [];
+
+  await loyaltyProgramDoc.append('collectionRules', collectionRulesData[0]);
+
+  await loyaltyProgramDoc.append('collectionRules', collectionRulesData[1]);
+
+  await loyaltyProgramDoc.sync();
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.LoyaltyProgram, loyaltyProgramData.name),
+    `Loyalty Program '${loyaltyProgramData.name}' exists `
+  );
+
+  const partyDoc = (await fyo.doc.getDoc(
+    ModelNameEnum.Party,
+    partyData.name
+  )) as Party;
+
+  partyDoc.loyaltyProgram = loyaltyProgramData.name;
+
+  await partyDoc.sync();
+
+  t.equals(
+    partyDoc.loyaltyProgram,
+    loyaltyProgramData.name,
+    `Loyalty Program '${loyaltyProgramData.name}' successfully added to Party '${partyData.name}'`
+  );
+});
+
+async function loyaltyPointEntryDoc(sinvName: string) {
+  const loyaltyPointEntryData = (await fyo.db.getAll(
+    ModelNameEnum.LoyaltyPointEntry,
+    {
+      fields: ['name', 'customer', 'loyaltyPoints', 'loyaltyProgramTier'],
+      filters: { invoice: sinvName! },
+    }
+  )) as {
+    name?: string;
+    customer?: string;
+    loyaltyPoints?: number;
+    loyaltyProgramTier?: string;
+  }[];
+
+  if (loyaltyPointEntryData) {
+    return loyaltyPointEntryData[0];
+  }
+}
+
+async function createSalesInvoice() {
+  const sinvDoc = fyo.doc.getNewDoc(ModelNameEnum.SalesInvoice, {
+    account: 'Debtors',
+    party: partyData.name,
+    items: [
+      {
+        item: itemData.name,
+        rate: itemData.rate,
+        quantity: 1,
+      },
+    ],
+  }) as SalesInvoice;
+
+  return sinvDoc;
+}
+
+test('create Sales Invoice and verify loyalty points are created correctly', async (t) => {
+  const sinvDoc = await createSalesInvoice();
+
+  await sinvDoc.sync();
+  await sinvDoc.runFormulas();
+  await sinvDoc.submit();
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.SalesInvoice, sinvDoc.name),
+    `Sales Invoice '${sinvDoc.name}' exists`
+  );
+
+  t.ok(
+    fyo.db.exists(ModelNameEnum.SalesInvoice, sinvDoc.loyaltyProgram),
+    `Loyalty Program '${sinvDoc.loyaltyProgram}' should be linked to Sales Invoice '${sinvDoc.name}'`
+  );
```

Comment:
wrong check. should be checking if Loyalty Program field has the expected value in the Sales Invoice document